File: src/test/java/com/raychenon/leetcode/StringCompressionTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 1/3/2020
 * https://leetcode.com/problems/string-compression/
 */
public class StringCompressionTest {


    private static StringCompression stringCompression;

    @BeforeClass
    public static void setUp() {
        stringCompression = new StringCompression();
    }

    @Test
    public void aabbcccTest() {
        assertEquals(6, stringCompression.compress(new char[]{'a', 'a', 'b', 'b', 'c', 'c', 'c'}));
    }

    @Test
    public void aabbaaaTest() {
        assertEquals(6, stringCompression.compress(new char[]{'a', 'a', 'b', 'b', 'a', 'a', 'a'}));
    }

    @Test
    public void aabbaaadeTest() {
        assertEquals(8, stringCompression.compress(new char[]{'a', 'a', 'b', 'b', 'a', 'a', 'a', 'd', 'e'}));
    }

    @Test
    public void moreThan10TimesTest() {
        // 11 times, "z10"
        assertEquals(3, stringCompression.compress(new char[]{'z', 'z', 'z', 'z', 'z', 'z', 'z', 'z', 'z', 'z'}));
    }


    @Test
    public void aTest() {
        assertEquals(1, stringCompression.compress(new char[]{'a'}));
    }

    @Test
    public void emptyCharTest() {
        assertEquals(0, stringCompression.compress(new char[]{}));
    }


}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/StringCompressionTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * User: raychenon
 * Date: 26/2/2020
 * https://leetcode.com/problems/string-compression/
 */
class StringCompressionTest {

    @Test
    fun aabbcccTest() {
        assertEquals(6, StringCompression.compress(charArrayOf('a', 'a', 'b', 'b', 'c', 'c', 'c')))
    }

    @Test
    fun aabbaaaTest() {
        assertEquals(6, StringCompression.compress(charArrayOf('a', 'a', 'b', 'b', 'a', 'a', 'a')))
    }

    @Test
    fun aabbaaadeTest() {
        assertEquals(8, StringCompression.compress(charArrayOf('a', 'a', 'b', 'b', 'a', 'a', 'a', 'd', 'e')))
    }

    @Test
    fun moreThan10TimesTest() {
        // 11 times, "z10"
        assertEquals(3, StringCompression.compress(charArrayOf('z', 'z', 'z', 'z', 'z', 'z', 'z', 'z', 'z', 'z')))
    }


    @Test
    fun aTest() {
        assertEquals(1, StringCompression.compress(charArrayOf('a')))
    }

    @Test
    fun emptyCharTest() {
        assertEquals(0, StringCompression.compress(charArrayOf()))
    }

}